Automattic/crowdsignal-forms
# add/crowdsignal-embed block

This is a new block based off the core embed block specially designed to embed Crowdsignal surveys.  Links and buttons have been tailored to Crowdsignal surveys.  The block will also only accept known Crowdsignal domains to avoid confusion.

The core functionality of the embed block largely remains the same and the final result is output to the format from the embed block so the native embed process takes over the duties of displaying the embed in the published page.

There is a limitation in the core embed block where the preview only renders to a certain height in the editor, that also exists in this block.

Testing instructions:
Checkout branch and build the plugin.

Verify "Survey" item exists in the block library and shows up when you search for a block via typing / or hitting the inline + button

Try embedding a non-crowdsignal URL like YouTube, verify the error
Try putting in random garbage text, verify the error
Try a valid Survey URL from the survey's sharing settings, confirm the embed preview loads
Preview the page, verify the embed shows up.
